clients/src/main/java/org/apache/kafka/common/protocol/Protocol.java

```diff
@@ -201,7 +201,7 @@
                                                                                          INT64,
                                                                                          "Message offset to be committed."),
                                                                                new Field("metadata",
-                                                                                         STRING,
+                                                                                         NULLABLE_STRING,
```

Comment:
What about OFFSET_COMMIT_REQUEST_PARTITION_V0 and OFFSET_COMMIT_REQUEST_PARTITION_V1?